src/extension/nodes/CompoundCloudPlane.cpp

```diff
@@ -0,0 +1,479 @@
+                        }
+                    }
+                    float avg = sum / count;
+                    float new_density = density[x][y][i] + (avg - density[x][y][i]) * diffusion_rate;
+                    old_density[x][y][i] = new_density * (1.0f - (1.0f - decay_rates[i]) * delta);
+                    
+                    if (old_density[x][y][i] < minimum_visible_density)
+                        old_density[x][y][i] = 0;
+                }
+                else {
+                    old_density[x][y][i] = 0;
+                }
+            }
+        }
+    }
+    texture_needs_update = true;
+}
+
+void CompoundCloudPlane::advect(float delta) {
+    std::lock_guard<std::mutex> lock(density_mutex);
+    std::swap(density, old_density);
+
+    float flow_speed = 0.05f * delta; 
+    for (int x = 0; x < size; ++x) {
+        for (int y = 0; y < size; ++y) {
+            float new_x = std::fmod(static_cast<float>(x) + flow_speed, static_cast<float>(size));
+            float new_y = std::fmod(static_cast<float>(y) + flow_speed, static_cast<float>(size));
+            int nx = static_cast<int>(new_x);
+            int ny = static_cast<int>(new_y);
+            
+            // Bilinear interpolation
+            float fx = new_x - nx;
+            float fy = new_y - ny;
+            int nx1 = (nx + 1) % size;
+            int ny1 = (ny + 1) % size;
+            
+            for (int i = 0; i < CLOUDS_IN_ONE; ++i) {
+                if (old_density[x][y][i] > minimum_visible_density) {
+                    float interpolated_density = 
+                        old_density[x][y][i] * ((1-fx)*(1-fy)) +
+                        old_density[nx1][y][i] * (fx*(1-fy)) +
+                        old_density[nx][ny1][i] * ((1-fx)*fy) +
+                        old_density[nx1][ny1][i] * (fx*fy);
+                    
+                    density[nx][ny][i] = interpolated_density;
+                }
+                else {
+                    density[nx][ny][i] = 0;
+                }
+            }
+        }
+    }
+    texture_needs_update = true;
+}
+
+bool CompoundCloudPlane::has_compounds() const {
+    std::lock_guard<std::mutex> lock(density_mutex);
+    for (const auto& row : density) {
+        for (const auto& cell : row) {
+            if (cell.length_squared() > minimum_visible_density * minimum_visible_density) {
+                return true;
+            }
+        }
+    }
+    return false;
+}
+
+void CompoundCloudPlane::update_texture() {
+    texture_needs_update = true;
+}
+
+godot::PackedInt32Array CompoundCloudPlane::get_compounds_array() const {
+    return compounds;
+}
+
+int CompoundCloudPlane::get_compounds_count() const {
+    return static_cast<int>(compounds.size());
+}
+
+void CompoundCloudPlane::update_compound_colors() {
+    // Define a map of compound IDs to their corresponding colors from compounds.json
+    std::unordered_map<int, godot::Color> compound_color_map = {

```

Comment:
Is this temporary code or? If this is not temporary this definitely needs to be sent by the C# side to configure the cloud correctly.